sources/models/src/modeled_types/shared.rs

```diff
@@ -250,3 +252,99 @@ mod test_url {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// FriendlyVersion represents a valid version string that can be prefixed with or without 'v.
+/// It can also be set to 'latest' in represent the latest version. It stores the original string
+/// and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct FriendlyVersion {
```

Comment:
Maybe `VersionTag` is a better name for this? But then this is also more restrictive than what you would expect for a revision tag.

Reply:
If "latest" is the only non sem-ver string you care about, you could have something like:

```rust
pub enum Version {
    SemVer(semver::Version),
    Latest,
}
```

Not a required change, just an idea.
Edit, I just saw your comment about without or without 'v', so perhaps not if you need the original string...